### PR TITLE
Swift 3.0

### PR DESCRIPTION
**Pull request by asynchrony-ringo**
Updated KitchenSink application to be based on the changes in the swift3 branch of the SparkSDK.

### CODE REVIEW

File: KitchenSink/AppFeedbackTableViewController.swift [hunk 1]
```diff
@@ -48,34 +48,34 @@ class AppFeedbackTableViewController: UITableViewController, MFMailComposeViewCo
     
     // MARK: - UI views
     
-    override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
-        if indexPath.section == 0 && indexPath.row == 1 {
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        if (indexPath as NSIndexPath).section == 0 && (indexPath as NSIndexPath).row == 1 {
```
Comment (xu0101) [inline KitchenSink/AppFeedbackTableViewController.swift]: Why need to convert to NSIndexPath?

File: KitchenSink/Assets.xcassets/AppIcon.appiconset/Contents.json [hunk 2]
```diff
@@ -1,6 +1,16 @@
 {
   "images" : [
     {
+      "idiom" : "iphone",
```
Comment (xu0101) [inline KitchenSink/Assets.xcassets/AppIcon.appiconset/Contents.json:4]: why add the block of code?

File: KitchenSink/Utils.swift [hunk 3]
```diff
@@ -37,51 +37,55 @@ class Utils {
...
     }
     
-    static func getDataFromUrl(urlString:String, completion: ((data: NSData?, response: NSURLResponse?, error: NSError? ) -> Void)) {
-        let url = NSURL(string: urlString)
-        NSURLSession.sharedSession().dataTaskWithURL(url!) { (data, response, error) in
-            completion(data: data, response: response, error: error)
-            }.resume()
+    static func getDataFromUrl(_ urlString:String, completion: @escaping ((_ data: Data?, _ response: URLResponse?, _ error: Error? ) -> Void)) {
+        let url = URL(string: urlString)
+
```
Comment (xu0101) [inline KitchenSink/Utils.swift]: Seems there are many redundant spaces ?

File: Podfile [hunk 4]
```diff
@@ -1,12 +1,10 @@
 source 'https://github.com/CocoaPods/Specs.git'
 
-platform :ios, '8.0'
+# platform :ios, '9.0'
```
Comment (xu0101) [inline Podfile]: Why do we comment it?

File: Podfile [hunk 5]
```diff
@@ -1,12 +1,10 @@
...
 
 target 'KitchenSink' do
-
-  pod 'SparkSDK'
-  pod 'Toast-Swift', '~> 1.1.0'
-  pod 'Cosmos', '~> 1.2'
-
+	pod 'SparkSDK', :git => 'https://github.com/ciscospark/spark-ios-sdk.git', :branch => 'swift3'
+	pod 'Cosmos', '~> 6.0.0'
+	pod 'Toast-Swift', git: 'https://github.com/scalessec/Toast-Swift.git', :tag => '2.0.0'
```
Comment (xu0101) [inline Podfile]: why is not ":git => 'https://github.com/scalessec/Toast-Swift.git'"?  It's different from the first line usage.

**Comment by xu0101** [comment]
Hope we can test all the main functionality of KitchenSink App, to make sure swift3.0 migration doesn't break SDK and its example.

File: Podfile (hunk 5 above)
Comment (asynchrony-ringo) [reply Podfile]: At the time we made the pull request Toast-Swift didn't have their release pushed to CocoaPods yet, but had the tag. They have the release out now, so we can change that.

File: KitchenSink/Assets.xcassets/AppIcon.appiconset/Contents.json (hunk 2 above)
Comment (asynchrony-ringo) [reply KitchenSink/Assets.xcassets/AppIcon.appiconset/Contents.json:4]: I'm not sure. I'm sure we didn't do it manually. Part of the project upgrade process perhaps?

**Review by xu0101 — APPROVED** [review]
Looks good to me

**Pull request merged**